Complete List Command

apollo-tan opened the pull request:
Fixes #83

apollo-tan committed:
Add Client Type information to the ClientCard display

apollo-tan committed:
Add support for filtering based on Clients, Buyers and Sellers

apollo-tan committed:
Update ModelManager and Model to support display of properties

- Add getFilteredPropertyList method to ModelManager and Model

apollo-tan committed:
Update UI to support for display of properties
- Update LogicManager and Logic to support fetch of getFilteredPropertyList
- Add components and controllers to support display of properties
- Temporary configure MainWindow to only display the properties

apollo-tan committed:
Add full support for list feature
- Modify UI and Model to support toggling between display of Clients and Properties based on user command
- Modify execute logic of ListCommand to support the display toggling

apollo-tan committed:
Refactor ListCommand logic to have more OOP

- Convert ListCommand to an abstract class and create subclasses to handle the actual logic of listing instead
- Update ListCommandParser to reflect this change

apollo-tan committed:
Update Test Cases in ListCommandTest.java

apollo-tan committed:
Fix styling issues and add java docs

apollo-tan committed:
Merge branch 'master' into branch-continue-list-command

apollo-tan committed:
Modify text in PropertyListCard to be consistent with ClientListCard

apollo-tan committed:
Add test cases to uncovered code

apollo-tan committed:
Add additional tests for ModelManager

yongkheehou commented on src/main/java/seedu/address/logic/commands/ListClientsCommand.java:
```diff
@@ -0,0 +1,28 @@
+        requireNonNull(model);
+
+        // Logic to list clients (i.e. buyers and sellers)
+        model.updateFilteredClientList(client -> true);
```
hey can we create a method to list the clients instead of passing in a conditional that always evaluates to `true`?

yongkheehou commented on src/main/java/seedu/address/logic/commands/ListPropertiesCommand.java:
```diff
@@ -0,0 +1,36 @@
+        model.setDisplayProperties();
+
+        // Print a message to indicate properties are being listed
+        System.out.println("Listing all properties");
```
try to use the logger instead of inserting print statements in the execution of the code?

yongkheehou commented on src/main/java/seedu/address/model/ModelManager.java:
```diff
@@ -267,19 +271,33 @@ public ReadOnlyPropertyBook getPropertyBook() {
-
     /**
-     * Returns an unmodifiable view of the list of {@code Person} backed by the internal list of
+     * Returns an unmodifiable view of the list of {@code Property} backed by the internal list of
```
nice catch haha

yongkheehou commented on src/main/java/seedu/address/ui/ClientCard.java:
```diff
@@ -43,6 +44,7 @@ public ClientCard(Client client, int displayedIndex) {
         this.client = client;
         id.setText(displayedIndex + ". ");
         name.setText(client.getName().fullName);
+        clientType.setText("[" + client.getTypeString() + "]");
```
wondering if this has been tested? how does it look on the UI?

apollo-tan replied:
The UI is currently like this. Let me know if you want any changes!

![image](https://github.com/user-attachments/assets/87359ff7-7e2b-456a-a710-ba6ac789b410)

![image](https://github.com/user-attachments/assets/470a9084-9ec0-4940-9e72-d5295f75c0f3)

apollo-tan committed:
Refactor code to look neater

- Remove unnecessary logs
- Modify execute command slightly to make more readable

yongkheehou reviewed (approved):
LGTM
